src/qt/transactionrecord.cpp

```diff
@@ -163,8 +164,31 @@ QList<TransactionRecord> TransactionRecord::decomposeTransaction(const CWallet *
                 if (ExtractDestination(txout.scriptPubKey, address))
                 {
                     // Sent to Bitcoin Address
-                    sub.type = TransactionRecord::SendToAddress;
-                    sub.address = CBitcoinAddress(address).ToString();
```

Comment:
// Sent to testnet or mainnet vote / beacon address
                    if (sub.address == GetBurnAddress())
                    {
                        std::vector<std::pair<std::string, std::string>> vTxNormalInfoIn = GetTxNormalBoincHashInfo(wtx);
                        for (const auto& vTxNormalInfo : vTxNormalInfoIn)
                        {
                            if (vTxNormalInfo.first == "Message Type")
                            {
                                if (vTxNormalInfo.second == "Vote" || vTxNormalInfo.second == "Add Poll")
                                {
                                    if (fDebug)
                                        OutputDebugStringF("Vote/Add Poll\n");

                                    sub.type = TransactionRecord::Vote;

                                    break;
                                }

                                else if (vTxNormalInfo.second == "Add Beacon Contract")
                                {
                                    sub.type = TransactionRecord::Beacon;

                                    break;
                                }
                                /* Message icon example as well
                                else if (vTxNormalInfo.second == "Text Message")
                                {
                                    sub.type TransactionRecord::Message;

                                    break;
                                }
                                */
                                // Message type is not one we are checking for. end the for auto loop
                                else
                                    break;
                            }
                        }
                    }
                }
---

that is likely better @denravonska  may chime in that

Reply:
the Message icon part is if u add a message icon change as i asked